views.go

```diff
@@ -110,13 +108,12 @@ func (lk *logKeeper) createBuild(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	hasher := md5.New()
-	if _, err = hasher.Write([]byte(bson.NewObjectId().Hex())); err != nil {
+	newBuildId, err := model.NewBuildId(buildParameters.Builder, buildParameters.BuildNum)
```

Comment:
should we also check, in the case where `buildParameters.S3 == true` , that the build doesn't exist in S3?

Reply:
I think no? If we've made it past the code that checks if it's a dupe in the database, then we've already written something to the DB. I think for our sanity it's probably best to just overwrite what's in S3. 

Like long term we want to check what's in S3, but if we find out that there is in fact a dup in S3 that wasn't in the DB I feel like there's not a lot we can do? I could log an error I suppose?

Reply:
Maybe logging an error is a good idea either way. Do you think we should write the data or just log an error and refuse to write to S3?

Reply:
Ah, I missed that this is just going to be for the current stage (where we still write everything to the db). It seems we don't need to worry about it since we're guaranteed that we'll never get past the db dupe check if it's already been created.